### ISSUE TITLE
role-name[path] doesn't trigger in playbooks

### ISSUE THREAD

**Issue by MarkusTeufelberger** [ansible/ansible-lint#2972]
##### Summary

Role names in `roles: [...]` in playbooks should also conform to naming conventions

##### Issue Type

- Bug Report

##### OS / ENVIRONMENT

```console (paste below)
ansible-lint --version
ansible-lint 6.11.0 using ansible 2.14.1


```

- ansible installation method: OS package
- ansible-lint installation method: pipx

##### STEPS TO REPRODUCE

Consider the following playbook (you need to create an empty file `./roles/path/with/slash/tasks/main.yml` first):

```console (paste below)
---
- name: Demo
  hosts: all
  roles:
    - role: path/with/slash
  tasks:
    - name: Demo
      ansible.builtin.import_role:
        name: path/with/slash

```

It errors on line 7 (the import_role task) but does _not_ complain about the same role name on line 5.

##### Desired Behavior

Don't only lint import_role, but also the `roles` list that does the same, just not as a task.

##### Actual Behavior

ansible-lint only detects the invalid name in line 7, not in line 5 (it also fails to detect other naming violations such as capital letters etc. there).

**Comment by ssbarnea** [comment]
@MarkusTeufelberger Do you think you could make a PR for this one? I do not think it should be too hard.

**Comment by MarkusTeufelberger** [comment]
Probably something like this?

```
    def matchplay(self, file: Lintable, data: dict[str, Any]) -> list[MatchError]:
        if file.kind != "playbook":
            return []
        if "roles" in data:
            # Check here somehow for role names and match against regex
        return []
```

`matchplay` isn't documented on https://ansible-lint.readthedocs.io/custom-rules/#rule-definitions unfortunately, but I'll give it a try.

**Comment by MarkusTeufelberger** [comment]
https://github.com/ansible/ansible-lint/blob/main/test/rules/test_role_names.py seems to render out a very basic role + playbook in a very convoluted yet not very extensible way. Some other rules are tested within their definition file, what is the preferred way?

**Comment by ssbarnea** [comment]
Keep the same testing type that is used by the rule. Historically the embedded tests are newer, but migrating old ones was never a priority.
